modules/@apostrophecms/area/ui/apos/components/AposWidgetControls.vue

```diff
@@ -89,119 +92,64 @@ export default {
-      // Edit
-      if (!this.options.contextual) {
-        controls.push({
-          ...this.widgetDefaultControl,
-          label: 'apostrophe:edit',
-          icon: 'pencil-icon',
-          disabled: this.disabled,
-          tooltip: {
-            content: 'apostrophe:editWidget',
-            placement: 'left'
-          },
-          action: 'edit'
-        });
-      }
-
       // Custom widget operations displayed in the primary controls
-      controls.push(
-        ...this.widgetPrimaryOperations.map(operation => ({
+      return this.widgetPrimaryOperations.map(operation => {
+        const disabled = this.disabled ||
+            (operation.disabledIfData &&
+              checkIfConditions(this.$props, operation.disabledIfData));
+
+        return {
           ...this.widgetDefaultControl,
           ...operation,
-          disabled: this.disabled,
+          disabled,
           tooltip: {
-            content: operation.label,
+            content: !disabled && (operation.tooltip || operation.label),
             placement: 'left'
           }
-        }))
-      );
-
-      return controls.filter(control => (
-        !this.widgetSkipControlsMap.get(control.action) &&
-          !this.widgetSkipControlsMap.get(control.name)
-      ));
+        };
+      });
     },
     widgetSecondaryControls() {
-      const controls = [];
+      const renderOperation = (operation) => {
+        const disabled = this.disabled ||
+            (operation.disabledIfData &&
+            checkIfConditions(this.$props, operation.disabledIfData));
 
-      // Cut
-      controls.push({
-        label: 'apostrophe:cut',
-        icon: 'content-cut-icon',
-        action: 'cut'
-      });
+        return {
+          ...operation,
+          modifiers: [
+            ...disabled ? [ 'disabled' ] : []
+          ]
+        };
+      };
+      const controls = this.widgetNativeSecondaryOperations.map(renderOperation);
 
-      // Copy
-      controls.push({
-        label: 'apostrophe:copy',
-        icon: 'content-copy-icon',
-        action: 'copy'
-      });
+      if (!this.widgetCustomSecondaryOperations.length) {
+        return controls;
+      }
 
-      // Clone
       controls.push({
-        label: 'apostrophe:duplicate',
-        icon: 'content-duplicate-icon',
-        action: 'clone',
-        modifiers: [
-          ...(this.disabled || this.maxReached) ? [ 'disabled' ] : []
-        ]
+        separator: true
       });
 
-      if (this.widgetSecondaryOperations.length) {
-        controls.push({
-          separator: true
-        });
-      }
-
       // Custom widget operations displayed in the secondary controls
-      controls.push(...this.widgetSecondaryOperations);
-
-      return controls.filter(control => (
-        !this.widgetSkipControlsMap.get(control.action) &&
-          !this.widgetSkipControlsMap.get(control.name)
-      ));
+      return controls.concat(this.widgetCustomSecondaryOperations.map(renderOperation));
```

Comment:
use the same cb than for native, allow to disable custom secondary operations too.